builtin/providers/kubernetes/structures_pod.go

```diff
@@ -0,0 +1,494 @@
+			vl[i].HostPath = expandHostPathVolumeSource(v)
+		}
+		if v, ok := m["glusterfs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Glusterfs = expandGlusterfsVolumeSource(v)
+		}
+		if v, ok := m["nfs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].NFS = expandNFSVolumeSource(v)
+		}
+		if v, ok := m["rbd"].([]interface{}); ok && len(v) > 0 {
+			vl[i].RBD = expandRBDVolumeSource(v)
+		}
+		if v, ok := m["iscsi"].([]interface{}); ok && len(v) > 0 {
+			vl[i].ISCSI = expandISCSIVolumeSource(v)
+		}
+		if v, ok := m["cinder"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Cinder = expandCinderVolumeSource(v)
+		}
+		if v, ok := m["ceph_fs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].CephFS = expandCephFSVolumeSource(v)
+		}
+		if v, ok := m["fc"].([]interface{}); ok && len(v) > 0 {
+			vl[i].FC = expandFCVolumeSource(v)
+		}
+		if v, ok := m["flocker"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Flocker = expandFlockerVolumeSource(v)
+		}
+		if v, ok := m["flex_volume"].([]interface{}); ok && len(v) > 0 {
+			vl[i].FlexVolume = expandFlexVolumeSource(v)
+		}
+		if v, ok := m["azure_file"].([]interface{}); ok && len(v) > 0 {
+			vl[i].AzureFile = expandAzureFileVolumeSource(v)
+		}
+		if v, ok := m["vsphere_volume"].([]interface{}); ok && len(v) > 0 {
+			vl[i].VsphereVolume = expandVsphereVirtualDiskVolumeSource(v)
+		}
+		if v, ok := m["quobyte"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Quobyte = expandQuobyteVolumeSource(v)
+		}
+		if v, ok := m["azure_disk"].([]interface{}); ok && len(v) > 0 {
+			vl[i].AzureDisk = expandAzureDiskVolumeSource(v)
+		}
+		if v, ok := m["photon_persistent_disk"].([]interface{}); ok && len(v) > 0 {
+			vl[i].PhotonPersistentDisk = expandPhotonPersistentDiskVolumeSource(v)
+		}
+	}
+	return vl, nil
+}
+
+func expandPodTemplateSpec(p []interface{}) (v1.PodTemplateSpec, error) {
+	obj := v1.PodTemplateSpec{}
+	if len(p) == 0 || p[0] == nil {
+		return obj, nil
+	}
+	in := p[0].(map[string]interface{})
+	meta := expandMetadata(in["metadata"].([]interface{}))
+	obj.ObjectMeta = meta
+
+	if v, ok := in["spec"].([]interface{}); ok && len(v) > 0 {
+		var err error
+		obj.Spec, err = expandPodSpec(v)
+		if err != nil {
+			return obj, err
+		}
+	}
+	return obj, nil
+}
+
+func patchPodSpec(pathPrefix, prefix string, d *schema.ResourceData) (PatchOperations, error) {
+	ops := make([]PatchOperation, 0)
+	prefix += ".0."
+
+	if d.HasChange(prefix + "active_deadline_seconds") {
+
+		v := d.Get(prefix + "active_deadline_seconds").(int)
+		ops = append(ops, &ReplaceOperation{
+			Path:  pathPrefix + "/activeDeadlineSeconds",
+			Value: v,
+		})
+	}
+
+	if d.HasChange(prefix + "containers") {
+		containers := d.Get(prefix + "containers").([]interface{})
+		value, _ := expandContainers(containers)
+
+		for i, v := range value {
+			ops = append(ops, &ReplaceOperation{
+				Path:  pathPrefix + "/containers/" + strconv.Itoa(i) + "/image",
+				Value: v.Image,
+			})
+		}
+
```

Comment:
I think we can only update the _active_deadline_seconds_ and container's _image_ in a Pod.